### PR TITLE
Voice disconnecting

### PR DESCRIPTION
**Pull request by overlordbleck**
🚀

### CODE REVIEW

File: lua/cfc_voice_channels/core/sv/sv_channel_control.lua [hunk 1]
```diff
@@ -55,14 +55,15 @@ function cfc_voice:CreateChannel(caller, name, password)
         return 
     end
 
-    cfc_voice.Channels[table.Count(cfc_voice.Channels) + 1] = {
+    local i = table.Count(cfc_voice.Channels) + 1
+    cfc_voice.Channels[i] = {
```
Comment (sarahsturgeon) [inline lua/cfc_voice_channels/core/sv/sv_channel_control.lua:59]: This functionality could be extracted into its own method.

`storeChannel`, `saveChannel`, `trackChannel`, something like that

File: lua/cfc_voice_channels/core/sv/sv_channel_control.lua [hunk 2]
```diff
@@ -97,6 +98,18 @@ function cfc_voice:joinChannel(ply, channel)
     table.insert(channel.Users, ply)
 end
 
+function cfc_voice:ChannelPlayerDisconnect(channel)
```
Comment (sarahsturgeon) [inline lua/cfc_voice_channels/core/sv/sv_channel_control.lua]: Can this be renamed to `function cfc_voice:OnChannelPlayerDisconnect(channel)` ?

File: lua/cfc_voice_channels/core/sv/sv_player.lua [hunk 3]
```diff
@@ -10,4 +10,21 @@ function player:isInChannel()
...
+                return channel
+            end
+        end
+    end
+end
+
+hook.Add("PlayerDisconnected", "cfc_voice_on_disconnect", function(ply)
+    if not ply:isInChannel() then return end
+
+    local channel = ply:getPlayerChannel()
```
Comment (sarahsturgeon) [inline lua/cfc_voice_channels/core/sv/sv_player.lua]: I think this method name is too vague to add straight to the player table. `player:getPlayerVoiceChannel()` or even better, `player:getCFCVoiceChannel()`

**Review by sarahsturgeon — DISMISSED** [review]
A few things to look at

File: lua/cfc_voice_channels/core/sv/sv_channel_control.lua (hunk 1 above)
Comment (overlordbleck) [reply lua/cfc_voice_channels/core/sv/sv_channel_control.lua:59]: Ooh, ok i'll probably do that in a different PR maybe?

Comment (sarahsturgeon) [reply lua/cfc_voice_channels/core/sv/sv_channel_control.lua:59]: you can - it'd be really easy to do quickly

**Review by sarahsturgeon — APPROVED** [review]
👍

**Pull request merged**